source/features/tag-changelog-link.tsx

```diff
@@ -0,0 +1,114 @@
+	commit: string;
+	tag: string;
+}
+
+async function getNextPage(): Promise<DocumentFragment> {
+	const nextPageLink = select<HTMLAnchorElement>('.pagination a:last-child');
+	if (nextPageLink) {
+		return fetchDom(nextPageLink.href);
+	}
+
+	if (isSingleTagPage()) {
+		const [, tag = ''] = getRepoPath()!.split('releases/tag/', 2); // Already URL-encoded
+		return fetchDom(`/${getRepoURL()}/tags?after=${tag}`);
+	}
+
+	return new DocumentFragment();
+}
+
+async function init(): Promise<void | false> {
+	if (select.exists('.blankslate')) {
+		return false;
+	}
+
+	const tagRegex = /\/releases\/tag\/(.*)/;
+	const documents = [document, await getNextPage()] as any; // TODO: fix select-dom types to accept mixed arrays
+
+	// not(.js-timeline-tags-expander) is needed as there can be some collapsed tags
+	// See https://github.com/facebook/react/releases?after=v16.7.0 for an example
+	const tagContainerSelector = '.release, .Box-row .commit, .release-entry .release-main-section:not(.commit):not(.js-timeline-tags-expander)';
+
+	// These selectors need to work on:
+	// https://github.com/facebook/react/tags (tags list)
+	// https://github.com/facebook/react/releases (releases list)
+	// https://github.com/parcel-bundler/parcel/releases (releases list without release notes)
+	const tagSelector = 'h4.commit-title > a[href*="/releases/tag/"], div.release-header .f1 > a[href*="/releases/tag/"]';
+
+	const allTags: TagDetails[] = select.all(tagContainerSelector, documents).map(element => ({
+		element,
+		tag: decodeURIComponent(select<HTMLAnchorElement>(tagSelector, element)!.pathname.match(tagRegex)![1]),
+		commit: select('.commit ul a[href*="/commit/"].muted-link', element)!.textContent!.trim()
+	}));
+
+	for (const [index, container] of allTags.entries()) {
+		const previousTag = getPreviousTag(index, allTags);
+
+		if (previousTag !== false) {
+			const unorderedLists = select.all('.commit > ul.f6, .commit > .release-header > ul, div:first-child > ul', container.element);
+
+			for (const list of unorderedLists) {
+				list.append(
+					<li className={list.lastElementChild!.className}>
+						<a
+							className="muted-link tooltipped tooltipped-n"
+							aria-label={'See changes since ' + previousTag}
+							href={`/${getRepoURL()}/compare/${previousTag}...${allTags[index].tag}`}
+						>
+							{diff()} Changelog
+						</a>
+					</li>
+				);
+			}
+		}
+	}
+}
+
+const getPreviousTag = (index: number, allTags: TagDetails[]): string | false => {
+	let previousTag: string | false = false;
+
+	// If tag is `@parcel/integration-tests@1.12.2` then namespace is `@parcel/integration-tests`
+	const namespaceRegex = /@[^@]+$/;
+
+	for (let i = index + 1; i < allTags.length; i++) {
+		if (allTags[i].commit === allTags[index].commit) {
+			continue;
+		}
+
+		// Ensure that they have the same namespace. e.g. `parcel@1.2.4` and `parcel@1.2.3`
+		if (allTags[i].tag.split(namespaceRegex)[0] === allTags[index].tag.split(namespaceRegex)[0]) {
```

Comment:
Since `namespaceRegex` is a separate variable again, we might as well turn it into a global function, e.g.

```js
const getNameSpace = tag => tag.split(/@[^@]+$/)[0];
```

```suggestion
		if (getNameSpace(allTags[i].tag) === getNameSpace(allTags[index].tag)) {
```